lib/Sema/TypeCheckError.cpp

```diff
@@ -679,6 +679,18 @@ class ApplyClassifier {
   Classification classifyRethrowsArgument(Expr *arg, Type paramType) {
     arg = arg->getValueProvidingExpr();
 
+    // If this argument is `nil` literal, it doesn't cause the call to throw.
+    if (arg->isImplicit()) {
+      if (auto *DSCE = dyn_cast<DotSyntaxCallExpr>(arg)) {
+        if (auto *DE = dyn_cast<DeclRefExpr>(DSCE->getFn())) {
+          if (auto *EED = dyn_cast<EnumElementDecl>(DE->getDecl())) {
+            if (DE->isImplicit() && EED->getName().is("none"))
```

Comment:
Please just compare it against `Ctx.getOptionalNoneDecl()`.

Reply:
Ah sure, I didn't know that existed, sorry.